server/aws-lsp-yaml-json/patchYamlJsonPackage.js

```diff
@@ -35,7 +35,7 @@ const filePathToPatchPathUnsafeEval = {
 function applyPatch(filePathToPatchPath) {
     for (var filePath in filePathToPatchPath) {
         const pathToPatch = `${__dirname}/${filePathToPatchPath[filePath]}`
-        const script = `cd ${rootPackage} && patch ${filePath} ${pathToPatch}`
+        const script = `cd "${rootPackage}" && patch "${filePath}" "${pathToPatch}"`
         console.log(script)
         const output = execSync(script, { encoding: 'utf-8', timeout: 2000 })
```

Comment:
here is an untested (but probably correct) example. 

note the use of `cwd` to set the working directory.

```suggestion
        console.log(script);
        execFile('patch', [filePath, pathToPatch], { cwd: rootPackage, encoding: 'utf-8', timeout: 2000 }, (error, stdout, stderr)
```

Reply:
So what about the logging statement?  Would we want the following instead?

```js
function applyPatch(filePathToPatchPath) {
    for (var filePath in filePathToPatchPath) {
        const pathToPatch = `${__dirname}/${filePathToPatchPath[filePath]}`
        const patchProc = execFile('patch', [filePath, pathToPatch], { cwd: rootPackage, encoding: 'utf-8', timeout: 2000 })

        console.log({
            cmd: patchProc.spawnfile,
            args: patchProc.spawnargs,
            cwd: rootPackage,
        });
    }
}
```

Reply:
I've once again created the PR with the code from above here: https://github.com/aws/language-servers/pull/270

I've confirmed that `applyPatch` works as well.